--- a/sdk/src/lib.rs
+++ b/sdk/src/lib.rs
@@ -314,6 +314,7 @@
 		));
 
 		Ok(Arc::new(LoggedInSdk {
+			credentials,
 			user_facade,
 			entity_client,
 			service_executor,
@@ -474,6 +475,7 @@
 #[allow(dead_code)]
 #[derive(uniffi::Object)]
 pub struct LoggedInSdk {
+	credentials: Credentials,
 	user_facade: Arc<UserFacade>,
 	entity_client: Arc<EntityClient>,
 	service_executor: Arc<ResolvingServiceExecutor>,
@@ -490,6 +492,11 @@
 
 impl LoggedInSdk {
 	#[must_use]
+	pub fn credentials(&self) -> Credentials {
+		self.credentials.clone()
+	}
+
+	#[must_use]
 	pub fn get_service_executor(&self) -> &Arc<ResolvingServiceExecutor> {
 		&self.service_executor
 	}

