--- a/sdk/src/blobs/binary_blob_wrapper_serializer.rs
+++ b/sdk/src/blobs/binary_blob_wrapper_serializer.rs
@@ -1,6 +1,7 @@
 #[cfg(test)]
 use crate::id::generated_id::GENERATED_ID_BYTES_LENGTH;
 use crypto_primitives::key::GenericAesKey;
+#[cfg(test)]
 use thiserror::Error;
 
 ///
@@ -53,6 +54,7 @@
 	pub(crate) binary: Vec<u8>,
 }
 
+#[cfg(test)]
 #[derive(Error, Debug, uniffi::Error, Eq, PartialEq, Clone)]
 pub enum BinaryBlobWrapperSerializationError {
 	#[error("InvalidNumberOfBlobsError: expected: {expected}, actual: {actual}")]
--- a/sdk/src/instance_mapper.rs
+++ b/sdk/src/instance_mapper.rs
@@ -1288,7 +1288,7 @@
 }
 
 impl ElementValue {
-	fn as_unexpected(&self) -> Unexpected {
+	fn as_unexpected(&self) -> Unexpected<'_> {
 		match self {
 			ElementValue::Null => Unexpected::Other("null"),
 			ElementValue::String(v) => Unexpected::Str(v),
